app/gui/src/project-view/components/GraphEditor/widgets/WidgetPort.vue

```diff
@@ -143,6 +143,16 @@ export const widgetDefinition = defineWidget(
     score: (props, _db) => {
       const portInfo = injectPortInfo(true)
       const value = props.input.value
+      // This is a workaround to avoid automatic port around type annotated expressions
+      // in argument positions. A port needs to be created with portId override, and it
+      // will be handled by `WidgetTypeCastPort`. Without this check, the port will be
+      // created with invalid portId because of `ArgumentInfoKey` being set on the input.
+      if (
+        props.input.value instanceof Ast.TypeAnnotated &&
+        props.input.value.id === props.input.portId
+      ) {
+        return Score.Mismatch
+      }
```

Comment:
`props.input.value` is aliased just a line above.

Also, shouldn't this kind of problems be resolved by `WidgetTypeCastPort` having higher priority?

Reply:
Thanks for noting. 
> Also, shouldn't this kind of problems be resolved by WidgetTypeCastPort having higher priority?

It's not that easy because the priority requirements are quite nuts here. For example, `WidgetTypeCastPort` must match after `Group`, but `Group` has priority 999, and `Group` must have a lower priority than the `WidgetPort` in order not to break some common cases unrelated to type casting. So we have a strict requirement `Port < Group < WidgetTypeCastPort`, while we need `WidgetTypeCastPort < Port` to avoid this hack.

And I’m sure there are about a dozen similar unsolvable equations with other widgets; this one is the most obvious.